### PR TITLE
Login screen

### PR DESCRIPTION
**Pull request by idonov8**

### CODE REVIEW

File: App/Containers/Root/RootScreen.js [hunk 1]
```diff
@@ -1,37 +1,57 @@
...
   componentDidMount() {
-    // Run the startup saga when the application is starting
-    this.props.startup()
+    // Run the startup saga only if userId is available
+    if (this.props.userId !== null)
+       this.props.startup()
+  }
+
+  componentDidUpdate(prevProps) {
+    if (this.props.userId !== prevProps.userId) {
```
Comment (shacharbuda) [inline App/Containers/Root/RootScreen.js:19]: add another condition `&& this.props.userId` as if userId changed to null/undefinded somehow the app will enter startUp saga again

File: App/Containers/Root/RootScreen.js [hunk 2]
```diff
@@ -1,37 +1,57 @@
...
+        this.props.startup()
+    }
   }
 
   render() {
     return (
       <View style={Helpers.fill}>
-        <AppNavigator
+      {
+        this.props.userId === null ?
```
Comment (shacharbuda) [inline App/Containers/Root/RootScreen.js:29]: this could be just `this.props.userId ? ...`

File: App/Navigators/AppNavigator.js [hunk 3]
```diff
@@ -15,7 +14,7 @@ const StackNavigator = createStackNavigator(
     SplashScreen: SplashScreen,
     // Make main screen is ScanningScreen
     MainScreen: ScanningScreen,
-    NoPermissionsScreen,
+                NoPermissionsScreen,
```
Comment (shacharbuda) [inline App/Navigators/AppNavigator.js:17]: redundant space

File: App/Sagas/SamplesSaga.js [hunk 4]
```diff
@@ -70,17 +70,22 @@ export function* sampleDataOnce() {
...
     const wifiDataForSample = yield call(wifiService.getWifiDataForSample, wifi);
     const roomDataForSample = {
-      room_id: yield select(roomIdSelector)
+      room_id: null // For now tells the server this is a client and not a mapper.
+      // TODO: remove ASAP. This is ugly :D 
     };
     const phoneDataForSample = yield call(phoneService.getPhoneDataForSample);
-
+    const userDataForSample = {
+      userId: yield select(userIdSelector)
```
Comment (shacharbuda) [inline App/Sagas/SamplesSaga.js:78]: `user_id` for Ron

File: App/Containers/Root/RootScreen.js (hunk 2 above)
Comment (idonov8) [reply App/Containers/Root/RootScreen.js:29]: walla

**Pull request merged**